lib/elixir_internal_certificate_worker/scraper/user_search_worker.ex

```diff
@@ -0,0 +1,75 @@
+defmodule ElixirInternalCertificateWorker.Scraper.UserSearchWorker do
+  use Oban.Worker,
+    queue: :user_search,
+    max_attempts: 3,
+    unique: [period: 30]
+
+  alias ElixirInternalCertificate.Repo
+  alias ElixirInternalCertificate.Scraper.{ScraperEngine, Scrapers}
+  alias ElixirInternalCertificateWorker.Scraper.HtmlParsingHelper
+
+  @max_attempts 3
+
+  @impl Oban.Worker
+  def perform(%Oban.Job{args: %{"user_search_id" => user_search_id}, attempt: @max_attempts}) do
+    {:ok, _} = update_user_search(user_search_id, :failed)
+
+    {:error, "max attempts reached, attempt: #{@max_attempts}"}
```

Comment:
not sure about this, it seems like this job always returns the `{:error` state until it reaches the `max_attempt` regardless the result is a success or not?

Reply:
@andyduong1920, I've tried it out, seems it should be 4 (update to ```failed``` status in the 4th tried after 3 unsuccessful attempts is the right approach). But it is quite hard for others trying to read the source code, do you have any suggestions.